src/libraries/Common/src/System/IO/Archiving.Utils.Windows.cs

```diff
@@ -69,11 +69,8 @@ public static unsafe string EntryFromPath(ReadOnlySpan<char> path, bool appendPa
                     string.Empty;
             }
 
-#pragma warning disable CS8500 // takes address of managed type
-            ReadOnlySpan<char> tmpPath = path; // avoid address exposing the span and impacting the other code in the method that uses it
-            return string.Create(appendPathSeparator ? tmpPath.Length + 1 : tmpPath.Length, (appendPathSeparator, RosPtr: (IntPtr)(&tmpPath)), static (dest, state) =>
+            return string.Create(appendPathSeparator ? path.Length + 1 : path.Length, (appendPathSeparator, path), static (dest, state) =>
```

Comment:
Don't think `(appendPathSeparator, path)` going to work because `Tuple` can't have `ref struct` values. Not sure it's possible to expand `Tuple` to do that at this point.

Reply:
Yup. I missed the parens in this case when fixing up usage. Will revert.

> Not sure it's possible to expand Tuple to do that at this point.

I don't think it's possible without additional language/runtime features that allow the refstruct-ness of ValueTuple to be based off of whether any of its Ts are ref struct.

We could of course introduce RefStructValueTuples, and associated language support, but I don't think that's a conversation to have right now...